File: array/167.cpp
#include <bits/stdc++.h>

using namespace std;

bool containsDuplicate(vector<int>& nums) {
    unordered_set<int> freq;

    for(const auto& elem: nums) {
        auto item = freq.insert(elem);

        if(item.second == false)
            return true;
    }

    return false;

}

int main() {
    vector<int> nums {1,2,3,4,5,6,6, 10000};
    cout<<boolalpha<<containsDuplicate(nums);


    return 0;
}

File: array/167.py
from typing import List
from collections import Counter


def containsDuplicate(nums: List[int]) -> bool:
    freq = Counter(nums)

    for elem, count in freq.items():
        if count > 1:
            return True

    return False


print(containsDuplicate([1, 1, 1, 3, 3, 4, 3, 2, 4, 2]))
